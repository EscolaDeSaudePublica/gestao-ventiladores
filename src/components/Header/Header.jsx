import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Container } from '@material-ui/core';
import Menu from '../Menu/Menu';
import MENU_ROUTES from '../Menu/MenuConfig';

const useStyles = makeStyles(() => ({
  header: {
    background: '#4CAF50',
  },
  container: {
    height: 190,
    marginTop: 50,
    paddingLeft: 10,
  },
  title: {
    textTransform: 'uppercase',
    fontSize: 24,
    lineHeight: 1.3,
    color: '#fff',
    fontWeight: 'bold',
  },
}));

const Header = () => {
  const classes = useStyles();

  return (
    <header className={classes.header}>
      <Container className={classes.container}>
        <h1 data-testid="title" className={classes.title}>
          {'central '}
          <br />
          {'de '}
          <br />
          {'ventiladores '}
        </h1>
      </Container>
      <Menu menuRoutes={MENU_ROUTES} />
    </header>
  );
};

export default Header;
