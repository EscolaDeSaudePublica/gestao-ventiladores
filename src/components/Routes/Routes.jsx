import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import PropTypes from 'prop-types';
import { randomIndex } from '../../utils';
import IndexServiceOrder from '../../pages/SerrviceOrder';

export const hasRoutes = (routes) => {
  if (routes && routes.length > 0) {
    return true;
  }
  return false;
};

const Routes = ({ routes }) => (
  <BrowserRouter>
    <Switch>
      {hasRoutes(routes) ? routes.map((route) => (
        <Route key={randomIndex()} path={route.path} exact component={route.component} />
      )) : <Route key={randomIndex()} path="/" exact component={IndexServiceOrder} />}
    </Switch>
  </BrowserRouter>
);

Routes.propTypes = {
  routes: PropTypes.instanceOf(Array).isRequired,
};

export default Routes;
