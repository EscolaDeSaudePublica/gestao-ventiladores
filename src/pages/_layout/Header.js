import React from "react";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import AppBar from "@material-ui/core/AppBar";
import {makeStyles} from "@material-ui/core/styles";
import {Link} from "react-router-dom";

export default function Header () {
  const classes = useStyles();

  return (
    <header>
      <AppBar
        position="absolute"
        className={classes.appBar}
      >
        <Toolbar className={classes.toolbar}>
          <div className={classes.divTitle}>
            <Typography
              className={classes.text}
              noWrap
            >
              Central
              <br/> de
              <br/> Ventiladores
            </Typography>
          </div>
          <div className={classes.divUser}>
            <Typography
              className={classes.textUser}
              noWrap
            >
              {/*usuario, joe*/}
            </Typography>
          </div>
        </Toolbar>
        <div className={classes.appBarFooter}>
          <div className={classes.divTextFooter}>
            <Link
              to="/"
              className={classes.link}
            >
              + Triagem
            </Link>

            <Link
              to="/Diagnostico"
              className={classes.link}
            >
              + Diagnóstico
            </Link>

            <Link
              to="/Demand"
              className={classes.link}
            >
              + Demanda
            </Link>

            <Link
              to="/maintenance"
              className={classes.link}
            >+ Manuntenção</Link>
          </div>
        </div>
      </AppBar>
    </header>
  );
}

const useStyles = makeStyles((theme) => ({
  appBar: {
    position: "relative",
    height: 300,
    background: "#4CAF50",
  },
  toolbar: {
    marginTop: 20,
    justifyContent: "space-between",
    width: "auto",
    marginLeft: theme.spacing(2),
    marginRight: theme.spacing(2),
    [theme.breakpoints.up(1200 + theme.spacing(2) * 2)]: {
      width: 1200,
      marginLeft: "auto",
      marginRight: "auto",
    },
  },
  appBarFooter: {
    width: "100%",
    background: "#357a38",
    height: 60,
    position: "absolute",
    bottom: 0,
  },
  divTextFooter: {
    height: 60,
    justifyContent: "space-evenly",
    display: "flex",
    marginLeft: theme.spacing(2),
    marginRight: theme.spacing(2),
    [theme.breakpoints.up(1200 + theme.spacing(2) * 2)]: {
      width: 1200,
      marginLeft: "auto",
      marginRight: "auto",
    },
    flexDirection: "row",
  },
  divTitle: {
    flexDirection: "row",
  },
  textUser: {
    fontSize: 20,
  },
  textFooter: {
    fontSize: 20,
    alignSelf: "center",
    color: "#fff",
  },
  text: {
    fontSize: 30,
    color: "#fff",
    fontWeight: "bold",
  },
  link: {
    fontSize: 18,
    alignSelf: "center",
    color: "#fff",
    textDecoration: "none",
  },
}));
