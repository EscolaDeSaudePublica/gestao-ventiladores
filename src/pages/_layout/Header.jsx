import React from 'react';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import AppBar from '@material-ui/core/AppBar';
import { makeStyles } from '@material-ui/core/styles';
import { Link } from 'react-router-dom';
import AssignmentSharpIcon from '@material-ui/icons/AssignmentSharp';
import SubtitlesSharpIcon from '@material-ui/icons/SubtitlesSharp';
import LibraryBooksSharpIcon from '@material-ui/icons/LibraryBooksSharp';
import LocalMallSharpIcon from '@material-ui/icons/LocalMallSharp';
import UnarchiveSharpIcon from '@material-ui/icons/UnarchiveSharp';
import AssignmentTurnedInSharpIcon from '@material-ui/icons/AssignmentTurnedInSharp';
import Container from '@material-ui/core/Container';
import { green } from '@material-ui/core/colors';
import Grid from '@material-ui/core/Grid';


export default function Header() {
  const classes = useStyles();

  const menuRoutes = [
    { label: 'ORDENS DE SERVIÇO', pathname: '/ordens-servicos', icon: <LibraryBooksSharpIcon font="small" /> },
    { label: 'TRIAGEM', pathname: '/triagens', icon: <AssignmentSharpIcon font="small" /> },
    { label: 'DIAGNÓSTICO', pathname: '/diagnosticos', icon: <SubtitlesSharpIcon font="small" /> },
    { label: 'DEMANDA', pathname: '/demandas', icon: <LocalMallSharpIcon font="small" /> },
    { label: 'CALIBRAGEM', pathname: '/calibragem', icon: <AssignmentTurnedInSharpIcon font="small" /> },
    { label: 'ENTREGA', pathname: '/entregas', icon: <UnarchiveSharpIcon font="small" /> },
  ];

  const currentPathname = window.location.pathname;

  return (
    <header>
      <AppBar
        position="absolute"
        className={classes.appBar}
      >
        <Toolbar className={classes.toolbar}>
          <div className={classes.divTitle}>
            <Typography
              className={classes.text}
              noWrap
            >
              <strong>
                CENTRAL
                <br />
                {' '}
                DE
                <br />
                {' '}
                VENTILADORES
              </strong>
            </Typography>
          </div>
          <div className={classes.divUser}>
            <Typography
              className={classes.textUser}
              noWrap
            >
              {/* usuario, joe */}
            </Typography>
          </div>
        </Toolbar>
        <div className={classes.appBarFooter}>
          <Container>
            <div className={classes.divTextFooter}>
              <Grid container alignItems="center" spacing={4}>
                {menuRoutes.map((item, index) => (
                  <Grid item key={index} className={currentPathname === item.pathname ? classes.currentLink : ''}>
                    <Link
                      to={item.pathname}
                      className={classes.link}
                    >
                      <div style={{ display: 'flex', alignItems: 'center' }}>
                        {item.icon}
                        {' '}
                        <span style={{ marginLeft: '0.5rem' }}>{item.label}</span>
                      </div>
                    </Link>
                  </Grid>
                ))}
              </Grid>
            </div>
          </Container>
        </div>
      </AppBar>
    </header>
  );
}

const useStyles = makeStyles((theme) => ({
  appBar: {
    position: 'relative',
    height: 300,
    background: green[500],
  },
  toolbar: {
    marginTop: 20,
    justifyContent: 'space-between',
    width: 'auto',
    marginLeft: theme.spacing(2),
    marginRight: theme.spacing(2),
    [theme.breakpoints.up(1200 + theme.spacing(2) * 2)]: {
      width: 1200,
      marginLeft: 'auto',
      marginRight: 'auto',
    },
  },
  appBarFooter: {
    width: '100%',
    background: green['800'],
    height: 60,
    position: 'absolute',
    bottom: 0,
  },
  divTextFooter: {
    height: 60,
    display: 'flex',
    marginLeft: theme.spacing(2),
    marginRight: theme.spacing(2),
    [theme.breakpoints.up(1200 + theme.spacing(2) * 2)]: {
      width: 1200,
      marginLeft: 'auto',
      marginRight: 'auto',
    },
    flexDirection: 'row',
  },
  divTitle: {
    flexDirection: 'row',
  },
  textUser: {
    fontSize: 20,
  },
  textFooter: {
    fontSize: 20,
    // alignSelf: "center",
    color: '#fff',
  },
  text: {
    fontSize: 30,
    color: '#fff',
    fontWeight: 'bold',
  },
  link: {
    // marginRight: "2rem",
    fontSize: 18,
    // alignSelf: "center",
    color: '#fff',
    textDecoration: 'none',
  },
  currentLink: {
    borderBottom: '4px white solid',
  },
}));
