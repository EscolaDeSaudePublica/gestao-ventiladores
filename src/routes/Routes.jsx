import React, { useContext } from 'react';
import {
  BrowserRouter,
  Route,
  Switch,
} from 'react-router-dom';
import { randomIndex } from '../utils';
import {
  PRIVATE_ROUTES_CONFIG,
  PUBLIC_ROUTES_CONFIG,
} from './routesConfig';
import AuthContext from '../contexts/AuthContext';

export const hasRoutes = (routes) => {
  if (routes && routes.length > 0) {
    return true;
  }
  return false;
};

const Routes = () => {
  const { loginStatus } = useContext(AuthContext);
  const routes = loginStatus ? PRIVATE_ROUTES_CONFIG : PUBLIC_ROUTES_CONFIG;
  if (!hasRoutes(routes)) {
    throw new Error('Routes component require routes as props');
  }
  return (
    <BrowserRouter>
      <Switch>
        {routes.map((route) => (
          <Route
            key={randomIndex()}
            path={route.path}
            exact={route.exact}
            component={route.component}
          />
        ))}
      </Switch>
    </BrowserRouter>
  );
};

export default Routes;
