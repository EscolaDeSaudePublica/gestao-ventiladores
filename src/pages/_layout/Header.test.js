/* eslint-disable no-undef */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { create } from 'react-test-renderer';
import { Router } from 'react-router';
import { createBrowserHistory } from 'history';
import Routes from '../../components/Routes/Routes';
import getRoutes from '../../router';
import Header from './Header';


describe('Header component', () => {
  it('should render Header component without errors', () => {
    const history = createBrowserHistory();
    const Wrapper = () => (
      <>
        <Router history={history}>
          <Routes routes={getRoutes()} />
        </Router>
      </>
    );
    const component = create(<Wrapper />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
