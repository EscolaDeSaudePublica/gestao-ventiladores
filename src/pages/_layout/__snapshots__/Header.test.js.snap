// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render Header component without errors 1`] = `
<div
  className="makeStyles-layout-2"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-appBar-3 MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-4 MuiToolbar-gutters"
    >
      <p
        className="MuiTypography-root makeStyles-text-5 MuiTypography-body1 MuiTypography-noWrap"
      >
        <p>
          CENTRAL
        </p>
        <p>
          DE
        </p>
        <p>
          VENTILADORES
        </p>
      </p>
    </div>
    <div
      className="MuiContainer-root makeStyles-appBarFooter-6 MuiContainer-maxWidthLg"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
      >
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <a
            className="makeStyles-label-8"
            href="/ordens-servicos"
            onClick={[Function]}
          >
            <span>
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                font="small"
                viewBox="0 0 24 24"
              >
                <path
                  d="M4 6H2v16h16v-2H4V6zm18-4H6v16h16V2zm-3 9H9V9h10v2zm-4 4H9v-2h6v2zm4-8H9V5h10v2z"
                />
              </svg>
            </span>
            <span>
              ORDENS DE SERVIÇO
            </span>
          </a>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <a
            className="makeStyles-label-8"
            href="/triagens"
            onClick={[Function]}
          >
            <span>
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                font="small"
                viewBox="0 0 24 24"
              >
                <path
                  d="M21 3h-6.18C14.4 1.84 13.3 1 12 1s-2.4.84-2.82 2H3v18h18V3zm-9 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm2 14H7v-2h7v2zm3-4H7v-2h10v2zm0-4H7V7h10v2z"
                />
              </svg>
            </span>
            <span>
              TRIAGEM
            </span>
          </a>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <a
            className="makeStyles-label-8"
            href="/diagnosticos"
            onClick={[Function]}
          >
            <span>
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                font="small"
                viewBox="0 0 24 24"
              >
                <path
                  d="M22 4H2v16h20V4zM4 12h4v2H4v-2zm10 6H4v-2h10v2zm6 0h-4v-2h4v2zm0-4H10v-2h10v2z"
                />
              </svg>
            </span>
            <span>
              DIAGNÓSTICO
            </span>
          </a>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <a
            className="makeStyles-label-8"
            href="/demandas"
            onClick={[Function]}
          >
            <span>
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                font="small"
                viewBox="0 0 24 24"
              >
                <path
                  d="M21 6h-4c0-2.76-2.24-5-5-5S7 3.24 7 6H3v16h18V6zm-9-3c1.66 0 3 1.34 3 3H9c0-1.66 1.34-3 3-3zm0 10c-2.76 0-5-2.24-5-5h2c0 1.66 1.34 3 3 3s3-1.34 3-3h2c0 2.76-2.24 5-5 5z"
                />
              </svg>
            </span>
            <span>
              DEMANDA
            </span>
          </a>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <a
            className="makeStyles-label-8"
            href="/calibragem"
            onClick={[Function]}
          >
            <span>
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                font="small"
                viewBox="0 0 24 24"
              >
                <path
                  d="M21 3h-6.18C14.4 1.84 13.3 1 12 1s-2.4.84-2.82 2H3v18h18V3zm-9 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm-2 14l-4-4 1.41-1.41L10 14.17l6.59-6.59L18 9l-8 8z"
                />
              </svg>
            </span>
            <span>
              CALIBRAGEM
            </span>
          </a>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <a
            className="makeStyles-label-8"
            href="/entregas"
            onClick={[Function]}
          >
            <span>
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                font="small"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18.71 3H5.29L3 5.79V21h18V5.79L18.71 3zM14 15v2h-4v-2H6.5L12 9.5l5.5 5.5H14zM5.12 5l.81-1h12l.94 1H5.12z"
                />
              </svg>
            </span>
            <span>
              ENTREGA
            </span>
          </a>
        </div>
      </div>
    </div>
  </header>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg"
    style={
      Object {
        "padding": "5rem",
      }
    }
  >
    <h6
      className="MuiTypography-root MuiTypography-h6"
    >
      Carregando dados...
    </h6>
    <div
      aria-valuenow={0}
      className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
      role="progressbar"
    >
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
        style={
          Object {
            "transform": "translateX(-100%)",
          }
        }
      />
    </div>
  </div>
  <div
    className="makeStyles-appBar-223"
  >
    <div
      className="makeStyles-info-224"
    >
      <div
        className="makeStyles-coluna-227"
      >
        <p
          className="MuiTypography-root makeStyles-title-225 MuiTypography-body1 MuiTypography-noWrap"
        >
          Recebimento
        </p>
        <p
          className="MuiTypography-root makeStyles-text-226 MuiTypography-body1 MuiTypography-noWrap"
        >
          Cadastro
        </p>
        <p
          className="MuiTypography-root makeStyles-text-226 MuiTypography-body1 MuiTypography-noWrap"
        >
          Material
        </p>
      </div>
      <div
        className="makeStyles-coluna-227"
      >
        <p
          className="MuiTypography-root makeStyles-title-225 MuiTypography-body1 MuiTypography-noWrap"
        >
          Triagem
        </p>
        <p
          className="MuiTypography-root makeStyles-text-226 MuiTypography-body1 MuiTypography-noWrap"
        >
          Origem
        </p>
        <p
          className="MuiTypography-root makeStyles-text-226 MuiTypography-body1 MuiTypography-noWrap"
        >
          Modelo
        </p>
        <p
          className="MuiTypography-root makeStyles-text-226 MuiTypography-body1 MuiTypography-noWrap"
        >
          Situação
        </p>
        <p
          className="MuiTypography-root makeStyles-text-226 MuiTypography-body1 MuiTypography-noWrap"
        >
          Acessórios
        </p>
      </div>
      <div
        className="makeStyles-coluna-227"
      >
        <p
          className="MuiTypography-root makeStyles-title-225 MuiTypography-body1 MuiTypography-noWrap"
        >
          Diagnóstico
        </p>
        <p
          className="MuiTypography-root makeStyles-text-226 MuiTypography-body1 MuiTypography-noWrap"
        >
          Ckecklist
        </p>
        <p
          className="MuiTypography-root makeStyles-text-226 MuiTypography-body1 MuiTypography-noWrap"
        >
          Equipamentos
        </p>
        <p
          className="MuiTypography-root makeStyles-text-226 MuiTypography-body1 MuiTypography-noWrap"
        >
          Observações
        </p>
      </div>
      <div
        className="makeStyles-coluna-227"
      >
        <p
          className="MuiTypography-root makeStyles-title-225 MuiTypography-body1 MuiTypography-noWrap"
        >
          Avaliação
        </p>
        <p
          className="MuiTypography-root makeStyles-text-226 MuiTypography-body1 MuiTypography-noWrap"
        >
          Peças
        </p>
        <p
          className="MuiTypography-root makeStyles-text-226 MuiTypography-body1 MuiTypography-noWrap"
        >
          Acessórios
        </p>
      </div>
      <div
        className="makeStyles-coluna-227"
      >
        <p
          className="MuiTypography-root makeStyles-title-225 MuiTypography-body1 MuiTypography-noWrap"
        >
          Manuntenção
        </p>
        <p
          className="MuiTypography-root makeStyles-text-226 MuiTypography-body1 MuiTypography-noWrap"
        >
          Checklist
        </p>
        <p
          className="MuiTypography-root makeStyles-text-226 MuiTypography-body1 MuiTypography-noWrap"
        >
          Equipamento
        </p>
        <p
          className="MuiTypography-root makeStyles-text-226 MuiTypography-body1 MuiTypography-noWrap"
        >
          Observações
        </p>
      </div>
    </div>
    <div
      style={
        Object {
          "alignSelf": "center",
        }
      }
    >
      <p
        className="MuiTypography-root makeStyles-text-226 MuiTypography-body1 MuiTypography-noWrap"
      >
        © 2017 - 2020 – GOVERNO DO ESTADO DO CEARÁ. TODOS OS DIREITOS RESERVADOS - v1.1.5
      </p>
    </div>
  </div>
</div>
`;
